scarlet/source.py

```diff
@@ -295,6 +331,15 @@ def init_multicomponent_source(
 
     return seds, morphs, bbox
 
+class Random:
+    """ Class used to instantiate a RandomSource class from its kwargs.
+    """
+    def __init__(self, observations = None):
+        """Sets a *Source with all its arguments"""
+        self.kwargs = observations
+    def __call__(self, *args):
+        """Sets a *Source with all its arguments"""
+        return RandomSource(args[0], observation = self.kwargs)
```

Comment:
That is bad code because it serves almost no purpose. It simply provides a mechanism for you to run the external initialization, which is delayed from the actual instantiation of the source class. And its unmanageable again unless you want to force the user to write that empty code as well.

The delayed initialization (e.g. with a common coadd) can be a good idea, but this way isn't.

Reply:
Yes, it's the bit that I find inelegant. I could get rid of it, and only rely on the source classes, but it would not be able to support the new API anymore.

Reply:
Removed